apollo-router/src/plugins/telemetry/apollo_otlp_exporter.rs

```diff
@@ -88,24 +145,183 @@ impl ApolloOtlpExporter {
+            otlp_exporter,
+            errors_configuration: errors_configuration.clone(),
+            include_span_names,
+        });
+    }
+
+    pub(crate) fn prepare_for_export(&self, trace_spans: Vec<LightSpanData>) -> Vec<SpanData> {
+        let mut export_spans: Vec<SpanData> = Vec::new();
+
+        trace_spans.into_iter().for_each(|span| {
+            if span.attributes.get(&APOLLO_PRIVATE_REQUEST).is_some()
+                || self.include_span_names.contains(span.name.as_ref())
+            {
+                match span.name.as_ref() {
+                    SUBGRAPH_SPAN_NAME => export_spans.push(self.prepare_subgraph_span(span)),
+                    EXECUTION_SPAN_NAME => export_spans.push(self.prepare_execution_span(span)),
+                    SUBSCRIPTION_EVENT_SPAN_NAME => {
+                        if let Some(request_span) =
+                            self.synthesize_request_span_for_subscription_event(&span)
+                        {
+                            let child_span =
+                                self.prepare_subscription_event_span(span, &request_span);
+                            export_spans.push(request_span);
+                            export_spans.push(child_span);
+                        }
+                    }
+                    _ => export_spans.push(self.base_prepare_span(span)),
+                };
+            }
         });
+        export_spans
+    }
+
+    fn base_prepare_span(&self, span: LightSpanData) -> SpanData {
+        SpanData {
+            span_context: SpanContext::new(
+                span.trace_id,
+                span.span_id,
+                TraceFlags::default().with_sampled(true),
+                true,
+                TraceState::default(),
+            ),
+            parent_span_id: span.parent_span_id,
+            span_kind: span.span_kind.clone(),
+            name: span.name.clone(),
+            start_time: span.start_time,
+            end_time: span.end_time,
+            attributes: span.attributes,
+            events: EvictedQueue::new(0),
+            links: EvictedQueue::new(0),
+            status: span.status,
+            // TBD(tim): if the underlying exporter supported it, we could
+            // group by resource attributes here and significantly reduce the
+            // duplicate resource / scope data that will get sent on every span.
+            resource: Cow::Owned(self.resource_template.to_owned()),
+            instrumentation_lib: self.intrumentation_library.clone(),
+        }
+    }
+
+    /// Adds the "graphql.operation.subtype" attribute for subscription requests.
+    /// TBD(tim): we could do this for all OTLP?
+    /// or not do this at all and let the backend interpret?
+    fn prepare_execution_span(&self, mut span: LightSpanData) -> SpanData {
+        let op_type = span
+            .attributes
+            .get(&GRAPHQL_OPERATION_TYPE)
+            .and_then(extract_string)
+            .unwrap_or_default();
```

Comment:
Hmm - there are certain attributes that if you disable them it will not be able to be processed correctly in Studio.  This is one of them.  We might need to discuss the ways in which user configuration can break this.

Reply:
I guess the other point here is that it's only using this to decide whether or not to enrich with the subtype, so if the result is not `subscription` then we skip the logic and just export as usual.

Reply:
What if it overlaps with something from a spec or standard?  `graphql.operation.type` is documented in the semantic conventions and "recommended" https://opentelemetry.io/docs/specs/semconv/graphql/graphql-spans/

Reply:
I see another way to implement this which is maybe preferable - could we add the `apollo_private.operation.subtype` attribute in the `on_request` handler for the subscription requests?  e.g. https://github.com/apollographql/router/blob/bf29f974934e414d9c07ebdeadf6a5aa926221cf/apollo-router/src/plugins/telemetry/config_new/attributes.rs#L891-L894

Reply:
See https://github.com/apollographql/router/pull/4982/commits/78ea80d309c14a1d9a9e1d18a4c74d8d8c96ac1f